core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -176,90 +175,66 @@ public JournalStateMachine(Map<String, RaftJournal> journals, RaftJournalSystem
         () -> getLastAppliedTermIndex().getIndex() - mSnapshotLastIndex);
     MetricsSystem.registerGaugeIfAbsent(
         MetricKey.MASTER_JOURNAL_LAST_CHECKPOINT_TIME.getName(),
-        () -> mLastCheckPointTime);
+        () -> mLastSnapshotTime);
     MetricsSystem.registerGaugeIfAbsent(
         MetricKey.MASTER_JOURNAL_LAST_APPLIED_COMMIT_INDEX.getName(),
         () -> mLastAppliedCommitIndex);
     MetricsSystem.registerGaugeIfAbsent(
         MetricKey.MASTER_JOURNAL_CHECKPOINT_WARN.getName(),
         () -> getLastAppliedTermIndex().getIndex() - mSnapshotLastIndex
                 > Configuration.getInt(PropertyKey.MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES)
-                && System.currentTimeMillis() - mLastCheckPointTime > Configuration.getMs(
+                && System.currentTimeMillis() - mLastSnapshotTime > Configuration.getMs(
                 PropertyKey.MASTER_WEB_JOURNAL_CHECKPOINT_WARNING_THRESHOLD_TIME)
     );
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricKey.MASTER_EMBEDDED_JOURNAL_LAST_SNAPSHOT_DURATION_MS.getName(),
+        () -> mLastSnapshotDurationMs);
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricKey.MASTER_EMBEDDED_JOURNAL_LAST_SNAPSHOT_ENTRIES_COUNT.getName(),
+        () -> mLastSnapshotEntriesCount);
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricKey.MASTER_EMBEDDED_JOURNAL_LAST_SNAPSHOT_REPLAY_DURATION_MS.getName(),
+        () -> mLastSnapshotReplayDurationMs);
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricKey.MASTER_EMBEDDED_JOURNAL_LAST_SNAPSHOT_REPLAY_ENTRIES_COUNT.getName(),
+        () -> mLastSnapshotReplayEntriesCount);
```

Comment:
nit: can these be cached gauges? similar to all other gauges

Reply:
Which other gauges are cached? Also snapshotting happens at a specific interval of journal entries. Therefore, we cannot accurately predict when a new snapshot will be taken. I think it's important to have the most up-to-date information. Caching a gauge might mean a delay between the information being recorded and it being displayed.